### DISCUSSION TITLE
Packaging Risor scripts in an executable

### DISCUSSION THREAD

**Discussion by rubiojr** [deepnoodle-ai/risor#259] [Show and tell]
:wave: Hi!

I discovered Risor recently and I'm quickly becoming a big fan of it :sparkling_heart:.

To explore the language and framework and learn by doing, I've created [rsx](https://github.com/rubiojr/rsx), a tool built with Go and Risor itself that allows you to create Go binaries with Risor scripts embedded, so you can easily distribute your Risor tools.

It doesn't seem like the Risor binary offers that kind of functionality, or did I miss that?

I still don't know enough to provide better feedback, but I came across https://github.com/risor-io/risor/discussions/138 that would be quite nice.

Cheers :clinking_glasses:!

**Comment by myzie** [comment]
@rubiojr - this is great!

Something like this crossed my mind at some point, but I never started on it. Generally I think it would be handy to embed Risor scripts in this way.

I've only briefly reviewed your code so far, but the `embedImporter` seems like a great idea. I could imagine incorporating this or something similar into the core Risor library too.
